client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerLeakTest.java

```diff
@@ -38,7 +41,10 @@ public void call() {
             eventListener.expectOnlyCreatedEvent(1, cache(cacheName));
          }
       });
-      detectThreadLeaks(".*Client-Listener-ClientListenerLeakTest-.*");
+      WaitFor
+              .condition(() -> listThreadsMatchingRegexp(".*Client-Listener-ClientListenerLeakTest-.*").size() == 0)
+              .atMost(10, TimeUnit.MILLISECONDS)
+              .await();
```

Comment:
@slaskawi Ok, so here's my take on it: the last worker thread to finish changes the `ThreadPoolExecutor`'s state to `TERMINATED`, then notifies the thread that's blocked in `awaitTermination()`, and then runs a couple more instructions in `processWorkerExit()`. If the worker thread is preempted during those instructions, the thread leak detector will see it as live and the test will fail.

I've been discussing with @anistor and we think the proper way to make the test deterministic is to change `CacheNotifierListener.EventDispatcher.run()` to restore the old thread name in a finally block. Changing `stop()` to use `shutdownNow()` instead of `shutdown()` is optional :)